android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModule.java

```diff
@@ -151,12 +155,42 @@ public void onShouldStartLoadWithRequestCallback(final boolean shouldStart, fina
 
   @ReactMethod
   public void releaseWebView(final String webViewKey) {
-    // no-op for now
+    UiThreadUtil.runOnUiThread(() -> {
+      RNCWebView rncWebView = RNCWebViewMapManager.INSTANCE.getRncWebViewMap().get(webViewKey);
+      RNCWebViewManager.InternalWebView webView = (RNCWebViewManager.InternalWebView) RNCWebViewMapManager.INSTANCE.getInternalWebViewMap().get(webViewKey);
+
+      if (webView == null) {
+        FLog.w(TAG, "Failed to release webview with webViewKey: " + webViewKey);
+        return;
+      }
+
+      // Detach internal webview from the wrapper RNCWebView
+      if (rncWebView != null) {
+        RNCWebViewManager.InternalWebView internalWebView = rncWebView.detachWebView();
+        if (internalWebView != webView) {
+          throw new IllegalStateException("internalWebViewMap has a mismatched webview with key: " + webViewKey);
+        }
+      }
+
+      // Perform webview cleanup
+      if (webView.webViewKey != null || webView.keepWebViewInstanceAfterUnmount) {
```

Comment:
Should this one be an `&&` instead of an `||`? In https://github.com/discord/react-native-webview/pull/11/files#diff-e9700631fc9fa5600d2e1ee70eb2d28aa2a1fbc695b85f99981972d3904ca485R842, we're doing

`webView.webViewKey == null || !webView.keepWebViewInstanceAfterUnmount` . And applying a `!` / `not` to that (i.e. `!(webView.webViewKey == null || !webView.keepWebViewInstanceAfterUnmount)` would make that become `webView.webViewKey != null && webView.keepWebViewInstanceAfterUnmount`, right?

Reply:
yeah that's right = good call